Week3/homework/js-exercises/creditCardValidator.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+{
+  // In this project you'll write a script
+  // that validates whether or not a credit card number is valid.
+  // Here are the rules for a valid number:
+  function checkCard(cardNumber) {
+
+    // Number must be 16 digits
+    if (cardNumber.length !== 16) {
+      console.log(`Card must be 16 digits!`);
+      return false;
+    }
+
+    // All of them must be numbers
+    for (let item = 0; item < cardNumber.length; ++item) {
+      let cardNumberItem = cardNumber[item];
+      cardNumberItem = Number.parseInt(cardNumberItem);
+
+      if (Number.isInteger(cardNumberItem) !== true) {
+        console.log(`Must be numbers all of them!`)
+        return false;
+      }
+    }
+
+    // You must have at least two different digits represented
+    // (all of the digits cannot be the same)
+    const object = {};
+    let index = 0;
+    while (index < cardNumber.length) {
+      object[cardNumber[index]] = true;
+      ++index;
+    }
+    if (Object.keys(object).length < 2) {
+      console.log(`It must have at least two diffrent digits!`)
+      return false;
+    }
+
+
+
+    // The final digit must be even
+    if (cardNumber[cardNumber.length - 1] % 2 !== 0) {
```

Comment:
👍 or just divide the whole card number by 2 and check for remainder. If last digit is even or odd then the whole number would be even or odd respectively, right?